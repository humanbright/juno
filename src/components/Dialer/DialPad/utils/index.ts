export * from './DialPadUtils';
export * from './types';
export * from './useDialKeyboard';
export * from './useKeyAudio';
