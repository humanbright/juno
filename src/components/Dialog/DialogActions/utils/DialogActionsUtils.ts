import { RcClasses, UnitMap } from '../../../../foundation';
import { RcDialogSize } from '../../Dialog';
import { RcDialogActionsProps } from '../DialogActions';

export const RcDialogActionsClasses = RcClasses<RcDialogActionsProps>(
  [],
  'RcDialogActions',
);

export const RcDialogActionsSizes: UnitMap<RcDialogSize, number[]> = {
  xsmall: [2, 3.5, 4],
  small: [],
  medium: [],
  large: [],
  fullScreen: [],
};
