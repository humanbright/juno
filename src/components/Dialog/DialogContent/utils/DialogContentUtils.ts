import { RcClasses, UnitMap } from '../../../../foundation';
import { RcDialogSize } from '../../Dialog';
import { RcDialogContentProps } from '../DialogContent';

export const RcDialogContentClasses = RcClasses<RcDialogContentProps>(
  ['dividers'],
  'RcDialogContent',
);

export const RcDialogContentSpacings: UnitMap<RcDialogSize, number[]> = {
  xsmall: [0, 3.5, 3],
  small: [],
  medium: [],
  large: [],
  fullScreen: [],
};
