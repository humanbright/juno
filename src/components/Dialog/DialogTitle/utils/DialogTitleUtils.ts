import { RcClasses, UnitMap } from '../../../../foundation';
import { RcDialogSize } from '../../Dialog';
import { RcDialogTitleProps } from '../DialogTitle';

export const RcDialogTitleClasses = RcClasses<RcDialogTitleProps>(
  [],
  'RcDialogTitle',
);

export const RcDialogTitleSpacings: UnitMap<RcDialogSize, number[]> = {
  xsmall: [4, 3.5],
  small: [],
  medium: [],
  large: [],
  fullScreen: [],
};
