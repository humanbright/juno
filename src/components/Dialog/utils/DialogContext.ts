import { createContext, useContext } from 'react';

import { RcBaseSize, RcTheme } from '../../../foundation';

export type RcDialogChildrenSize = RcBaseSize<'small' | 'medium'>;

export type RcDialogChildrenProps = {
  /** size token */
  size?: RcDialogChildrenSize;
};

export type RcDialogContextValue = RcDialogChildrenProps;

export const RcDialogContext = createContext<RcDialogContextValue>({});

export const useRcDialogContext = () => useContext(RcDialogContext);

export const useDialogDefaultProps = ({
  theme,
  ...props
}: RcDialogChildrenProps & { theme?: RcTheme }) => {
  const context = useRcDialogContext();
  const { size = context.size || 'medium' } = props;

  return {
    ...props,
    size,
  };
};
