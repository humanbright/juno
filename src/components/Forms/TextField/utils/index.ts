export * from './OutlineTextFieldUtils';
export * from './TextFieldUtils';
