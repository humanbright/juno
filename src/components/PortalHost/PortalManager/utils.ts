import { UniqID } from './types';

type ResolveFn<T> = (value: T | PromiseLike<T>) => void;

export const createPromise = <T>(): [Promise<T>, ResolveFn<T>] => {
  let resolvePromise: ResolveFn<T> | null = null;
  const promise = new Promise<T>((resolve) => {
    resolvePromise = resolve;
  });
  return [promise, resolvePromise!];
};

export class UniqIdUtil {
  private _currentNumber = 0;

  get(): UniqID {
    return String(this._currentNumber++);
  }
}
