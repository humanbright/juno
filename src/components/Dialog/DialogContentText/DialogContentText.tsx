import MuiDialogContentText from '@material-ui/core/DialogContentText';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../../foundation';
import { RcTypographyProps } from '../../Typography';
import { TypographyStyle } from '../../Typography/styles';
import {
  MuiDefaultColor,
  RcCustomTypographyVariant,
} from '../../Typography/utils';
import { DialogContentTextStyle } from './styles';
import { RcDialogContentTextClasses } from './utils';

type RcDialogContentTextProps = {} & Pick<
  RcTypographyProps,
  'variant' | 'color' | 'weight' | 'component'
> &
  RcBaseProps<ComponentProps<typeof MuiDialogContentText>, 'variant' | 'color'>;

const _RcDialogContentText = forwardRef<any, RcDialogContentTextProps>(
  (inProps, ref) => {
    const props = useThemeProps({
      props: inProps,
      name: 'RcDialogContentText',
    });
    const {
      color: colorProp,
      component: componentProp,
      classes: classesProp,
      variant,
      weight,
      children,
      ...rest
    } = props;

    const component = componentProp || RcCustomTypographyVariant[variant!];

    const classes = useMemo(
      () => combineClasses(RcDialogContentTextClasses, classesProp),
      [classesProp],
    );

    const color = useMemo(
      () =>
        MuiDefaultColor.includes(colorProp!) ? (colorProp as any) : undefined,
      [colorProp],
    );

    return (
      <MuiDialogContentText
        // * for use html view variant
        data-variant={variant}
        // * for use html view color
        data-color={color}
        color={color}
        ref={ref}
        classes={classes}
        component={component}
        {...rest}
      >
        {children}
      </MuiDialogContentText>
    );
  },
);

const RcDialogContentText = styled(_RcDialogContentText)`
  ${DialogContentTextStyle}
  ${TypographyStyle}
`;

RcDialogContentText.defaultProps = {
  color: 'neutral.f04',
  variant: 'inherit',
};

RcDialogContentText.displayName = 'RcDialogContentText';

export { RcDialogContentText, RcDialogContentTextProps };
