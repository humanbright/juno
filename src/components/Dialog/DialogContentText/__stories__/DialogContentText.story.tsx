import { Meta, Story } from '@storybook/react';
import React, { ComponentProps } from 'react';

import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
  textPaletteChoice,
} from '../../../../storybook';
import { MuiDefaultColor } from '../../../Typography/utils';
import { RcDialogContentText } from '../DialogContentText';

export default {
  title: 'ðŸš€ Cleanup Components/Dialog/DialogContentText',
  component: RcDialogContentText,
  argTypes: {
    color: {
      control: {
        type: 'select',
        options: [...MuiDefaultColor, ...textPaletteChoice],
      },
    },
    ...sortInDocTable<keyof DialogContentTextProps>([
      'display',
      'align',
      'weight',
      'gutterBottom',
      'paragraph',
      'noWrap',
    ]),
    ...notControlInDocTable<keyof DialogContentTextProps>([]),
    ...notShowInDocTable<keyof DialogContentTextProps>([]),
  },
} as Meta;

type DialogContentTextProps = ComponentProps<typeof RcDialogContentText>;

export const DialogContentText: Story<DialogContentTextProps> = ({
  children,
  ...args
}) => {
  switchToControlKnobs();
  return <RcDialogContentText {...args}>{children}</RcDialogContentText>;
};

DialogContentText.storyName = 'DialogContentText';

DialogContentText.args = {
  children: 'You can set my maximum width and whether to adapt or not.',
};

DialogContentText.argTypes = {
  ...notControlInDocTable<keyof DialogContentTextProps>([]),
};

DialogContentText.parameters = {
  tags: [
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};
