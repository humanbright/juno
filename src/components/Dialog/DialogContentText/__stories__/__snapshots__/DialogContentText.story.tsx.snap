// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Dialog/DialogContentText DialogContentText 1`] = `
.c0 {
  font-family: Lato,Helvetica,Arial,sans-serif;
  color: #666666;
}

<p
  className="MuiTypography-root MuiDialogContentText-root c0 MuiTypography-body1"
  data-variant="inherit"
>
  You can set my maximum width and whether to adapt or not.
</p>
`;
